src/index/IndexImpl.Text.cpp

```diff
@@ -593,12 +616,23 @@ IdTable IndexImpl::readWordCl(
       textIndexFile_, [](WordIndex id) {
         return Id::makeFromWordVocabIndex(WordVocabIndex::make(id));
       });
-
-  textIndexReadWrite::readFreqComprList<Id, Score>(
-      idTable.getColumn(2).begin(), tbmd._cl._nofElements,
-      tbmd._cl._startScorelist,
-      static_cast<size_t>(tbmd._cl._lastByte + 1 - tbmd._cl._startScorelist),
-      textIndexFile_, &Id::makeFromInt);
+  if (textScoringMetric_ == TextScoringMetric::COUNT) {
+    textIndexReadWrite::readFreqComprList<Id, uint16_t>(
+        idTable.getColumn(2).begin(), tbmd._cl._nofElements,
+        tbmd._cl._startScorelist,
+        static_cast<size_t>(tbmd._cl._lastByte + 1 - tbmd._cl._startScorelist),
+        textIndexFile_, [](uint16_t score) {
+          return Id::makeFromInt(static_cast<uint64_t>(score));
+        });
+  } else {
+    textIndexReadWrite::readFreqComprList<Id, Score>(
+        idTable.getColumn(2).begin(), tbmd._cl._nofElements,
+        tbmd._cl._startScorelist,
+        static_cast<size_t>(tbmd._cl._lastByte + 1 - tbmd._cl._startScorelist),
+        textIndexFile_, [](Score score) {
+          return Id::makeFromDouble(static_cast<double>(score));
+        });
+  }
```

Comment:
1. How much sense does frequency encoding make with floating point BM25 values? 
   Is a generic compression (e.g. the zstandard we use elsewhere better).

2. If the answer is "NO", then there is quite some code duplication here.